fastapi_events/dispatcher.py

```diff
@@ -290,7 +292,7 @@ class UserCreated(pydantic.BaseModel):
                 )
 
         # OTEL
-        if payload and isinstance(payload, dict):
```

Comment:
Not sure why this type check was needed here, and whether injecting traceparent is needed for every payload or only some.

Reply:
IIRC, the `isinstance(payload, dict)` was added just in case people are using a custom class instance as payload. In this case, we should avoid injecting traceparent, since it most likely wont work.

Reply:
For the performance penalty, I see your point. I don't have any numbers on hand as for how much performance penalty it introduces, but I wouldn't mind us fixing it